src/main/java/com/github/boscojared/disparse/discord/Dispatcher.java

```diff
@@ -51,38 +52,54 @@ public void optionRequired(MessageReceivedEvent event, String message) {
         event.getChannel().sendMessage(message).queue();
     }
 
-    public void help(MessageReceivedEvent event, Command command, Collection<Flag> flags, Collection<Command> commands) {
+    private static final Comparator<CommandFlag> helpCommandFlagComparator = new Comparator<>() {
+        private Comparator<CommandFlag> comparator =
+            Comparator.comparing(this::shortNameKeyExtractor, Comparator.nullsLast(Comparator.naturalOrder()))
+                      .thenComparing(this::longNameKeyExtractor);
+
+        private Character shortNameKeyExtractor(CommandFlag flag) {
+            Character ch = flag.getShortName();
+            return ch == null ? null : Character.toLowerCase(ch);
+        }
+
+        private String longNameKeyExtractor(CommandFlag flag) {
+            return flag.getLongName().toLowerCase();
+        }
+
+        @Override
+        public int compare(CommandFlag o1, CommandFlag o2) {
+            return comparator.compare(o1, o2);
+        }
+    };
+
+    public void help(MessageReceivedEvent event, Command command, Collection<CommandFlag> flags, Collection<Command> commands) {
         EmbedBuilder builder = new EmbedBuilder();
         builder.setTitle(String.format("%s:  %s", command.getCommandName(), command.getDescription()))
                 .setDescription(String.format("Usage of command:  %s.  [+] may be repeated.", command.getCommandName()));
 
-        List<Flag> sortedFlags = flags.stream()
-                .sorted(
-                        Comparator.comparing((Flag flag) -> toLower(flag.getShortName()), Comparator.nullsLast(Comparator.naturalOrder()))
-                            .thenComparing(flag -> flag.getLongName().toLowerCase())
-                )
-                .collect(Collectors.toList());
 
-        List<Command> subcommands = commands.stream()
+        List<CommandFlag> sortedFlags = flags.stream().sorted(helpCommandFlagComparator).collect(Collectors.toList());
+
+        List<Command> subCommands = commands.stream()
```

Comment:
subcommand is one word so I don't think camelCase is necessary here

Reply:
agree, googled sub-command for some reason